File: Design-Patterns/StateMachine/Cpp/StateMachine/start_scene.cpp
#ifndef START_SCENE_H
#define START_SCENE_H

#include "scene.h"

start_scene::start_scene(scene_controller* controller) : scene(controller)
{
	
}

void start_scene::start()
{
	std::cout << "Start scene started" << std::endl;
}

void start_scene::update()
{
	std::cout << "Start scene updated" << std::endl;
}

void start_scene::end()
{
	std::cout << "Start scene ended" << std::endl;
}

#endif

File: Design-Patterns/StateMachine/CSharp/StateMachine/Scenes/StartScene.cs
using System;

namespace StateMachine.Scenes
{
    class StartScene : Scene
    {
        public StartScene(SceneController controller) : base(controller)
        {

        }

        public override void Start()
        {
            Console.WriteLine("Start scene started");
        }

        public override void Update()
        {
            Console.WriteLine("Start scene updated");
        }

        public override void End()
        {
            Console.WriteLine("Start scene ended");
        }
    }
}
